portfolio/src/main/webapp/script.js

```diff
@@ -117,19 +127,60 @@ function initMap() {
     content: contentString
   });
 
-  marker.addListener('click', function() {
-    if (marker.getAnimation() === google.maps.Animation.BOUNCE)  
-      marker.setAnimation(null);
-    else {
-      infowindow.open(map, marker);
-      marker.setAnimation(google.maps.Animation.BOUNCE);
+  marker.addListener("click", function() {
+    infowindow.open(map, marker);
+    marker.setAnimation(google.maps.Animation.BOUNCE);
       setTimeout(function() { marker.setAnimation(null); }, 600);
-    }
   });
+}
+
+var markers = [];
+var marker;
+
+function addPin(comment, time, i) {
+  setTimeout(function() {
+    marker = new google.maps.Marker({
+      position: {
+        lat: comment.lat, 
+        lng: comment.lng
+      }, 
+      animation: google.maps.Animation.DROP,
+      map: map
+    });
+    
+    markers.push(marker);
+
+    const contentString = "<div>" + comment.name + "</div>" + "<div>" + comment.text + "</div>";
+    const infowindow = new google.maps.InfoWindow({content: contentString});
+
+    marker.addListener("click", function() {
+      infowindow.open(map, marker);
+      marker.setAnimation(google.maps.Animation.BOUNCE);
+        setTimeout(function() { marker.setAnimation(null); }, 600);
+      });
+  }, time * i); 
+}
 
+function emptyMap() {
+  for (var i = 0; i < markers.length; i++) {
+    markers[i].setMap(null);
+  }
+  markers = [];   
+}
+
+function fillMap() {
+  emptyMap();
+  const number = document.getElementById("number").value;
+    fetch("/data?number=" + number).then(response => response.json()).then((comments) => {
+      var i = 0;
+      comments.forEach(c => {
```

Comment:
The callback you pass into `forEach` actually has an index parameter you can use so you can omit the local `i` var here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach#Parameters

Reply:
Didn't know that, thanks!